File: src/main/java/org/shark/example/service/permission/PermissionReadService.java
package org.shark.example.service.permission;

import lombok.extern.slf4j.Slf4j;
import org.shark.example.aop.permission.RequiresPermission;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class PermissionReadService {

    @RequiresPermission("admin:read") //FIXME
    public void start() {
        log.info("start work");
    }

}

File: src/main/kotlin/org/shark/example/service/permission/PermissionReadService.kt
package org.shark.example.service.permission

import org.shark.example.aop.permission.RequiresPermission
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class PermissionReadService {

    private val log = LoggerFactory.getLogger(PermissionReadService::class.java)

    @RequiresPermission("admin:read") //FIXME
    fun start() {
        log.info("start work")
    }
}
